x/ccv/provider/keeper/keeper.go

```diff
@@ -71,6 +72,59 @@ func NewKeeper(
 		evidenceKeeper:   evidenceKeeper,
 		feeCollectorName: feeCollectorName,
 	}
+
+	k.mustValidateFields()
+	return k
+}
+
+// Validates that the provider keeper is initialized with non-zero and
+// non-nil values for all its fields. Otherwise this method will panic.
+func (k Keeper) mustValidateFields() {
```

Comment:
Wondering in the future if it's worth refactoring to have `mustValidateFields` be generic for keepers, then pass in a set of validations instead per keeper....

```
validations := map[string]func(Keeper) bool{
	"number of fields in provider keeper is not 13": func(keeper Keeper) bool { return reflect.ValueOf(k).NumField() != 13 },
	"cdc is zero-valued or nil": func(k Keeper) bool { return reflect.ValueOf(k.cdc).IsZero() },
	"storeKey is zero-valued or nil": func(keeper Keeper) bool { return reflect.ValueOf(k.storeKey).IsZero()},
}

...

func (k Keeper) mustValidateFields(validations map[string]func(Keeper) bool) {
	for errMessage, failedValidation := range validations {
		if failedValidation(k) {
			panic(errMessage)
		}
	}
}
```

Reply:
This is a solid idea! Although in our repo there will only ever be 2 keepers defined, so maybe not worth adding the additional code and complexity? That sort of functionality living within the sdk repo would make the most sense to me.

Lmk if you strongly disagree and I can refactor in a future PR